src/ConstructiveSolidGeometry/SurfacePrimitives/ConeMantle.jl

```diff
@@ -76,33 +76,33 @@ radius_at_z(cm::ConeMantle{T,Tuple{T,T}}, z::T) where {T} = radius_at_z(cm.hZ, c
 get_φ_limits(cm::ConeMantle{T,<:Any,T}) where {T} = T(0), cm.φ
 get_φ_limits(cm::ConeMantle{T,<:Any,Nothing}) where {T} = T(0), T(2π)
 
-function normal(cm::ConeMantle{T,T,<:Any,:inwards}, pt::CartesianPoint{T}) where {T}
+function normal(cm::ConeMantle{T,T,<:Any,:inwards}, pt::CartesianPoint{T})::CartesianVector{T} where {T}
     pto = _transform_into_object_coordinate_system(pt, cm)
     cyl = CylindricalPoint(pto)
-    return CartesianVector(_transform_into_global_coordinate_system(
-            CartesianPoint(CylindricalPoint{T}(-cyl.r, cyl.φ, zero(T))), cm))
+    return _transform_into_global_coordinate_system(
+            CartesianVector(CartesianPoint(CylindricalPoint{T}(-cyl.r, cyl.φ, zero(T)))), cm)
```

Comment:
See this example: In order for this to work, the `CartesianPoint` is **first** converted to a `CartesianVector` and **then** transformed.

Reply:
> See this example: In order for this to work, the `CartesianPoint` is **first** converted to a `CartesianVector` and **then** transformed.

Maybe a silly question due to the fact that I am not very familiar with Julia but, wouldn't it be better to be explicit and define a new function _rotate_into_object_coordinate_system instead of adding a new method to _transform_into_object_coordinate_system where one depends on the order of parameters?

Reply:
We are using multiple dispatch here, meaning that the method for the function will be chosen by the **type** of the parameters, not their **order** (one is `_transform...(::CartesianPoint, ::AbstractPrimitive)`, the other is `_transform...(::CartesianVector, ::AbstractPrimitive)`, so the primitive is always second).

In that case, the programmer can use the same function name (and not care about if the first argument is a `Point` or a `Vector`) but it does different things depending on the type of the argument passed to the function.

Reply:
Thank you Felix,
it was indeed a silly question.
To attenuate (a bit) its silliness, I was referring to your starting replay
-    _transform_into_global_coordinate_system(es, CartesianVector{T}(zero(T), zero(T), one(T)))
+   _transform_into_global_coordinate_system(CartesianVector{T}(zero(T), zero(T), one(T)), es)
where the change of  order  implies the change of types that allows for multiple dispatch...
I'll stop interfering!
Dino